test/e2e/tests/test_model_package.py

```diff
@@ -0,0 +1,263 @@
+from e2e.common import config as cfg
+
+RESOURCE_PLURAL = "modelpackages"
+
+
+@pytest.fixture(scope="function")
+def xgboost_model_package_group():
+    resource_name = random_suffix_name("xgboost-model-package-group", 32)
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["MODEL_PACKAGE_GROUP_NAME"] = resource_name
+
+    (
+        model_package_group_reference,
+        model_package_group_spec,
+        model_package_group_resource,
+    ) = create_sagemaker_resource(
+        resource_plural=cfg.MODEL_PACKAGE_GROUP_RESOURCE_PLURAL,
+        resource_name=resource_name,
+        spec_file="xgboost_model_package_group",
+        replacements=replacements,
+    )
+    assert model_package_group_resource is not None
+
+    yield (model_package_group_reference, model_package_group_resource)
+
+    # Delete the k8s resource if not already deleted by tests
+    if k8s.get_resource_exists(model_package_group_reference):
+        _, deleted = k8s.delete_custom_resource(model_package_group_reference, 3, 10)
+        assert deleted
+
+
+@pytest.fixture(scope="function")
+def xgboost_versioned_model_package(xgboost_model_package_group):
+    resource_name = random_suffix_name("xgboost-versioned-model-package", 38)
+    (_, model_package_group_resource) = xgboost_model_package_group
+    print(model_package_group_resource)
+    model_package_group_resource_name = model_package_group_resource["spec"].get(
+        "modelPackageGroupName", None
+    )
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["MODEL_PACKAGE_GROUP_NAME"] = model_package_group_resource_name
+    replacements["MODEL_PACKAGE_RESOURCE_NAME"] = resource_name
+    reference, spec, resource = create_sagemaker_resource(
+        resource_plural=RESOURCE_PLURAL,
+        resource_name=resource_name,
+        spec_file="xgboost_versioned_model_package",
+        replacements=replacements,
+    )
+    assert resource is not None
+    if k8s.get_resource_arn(resource) is None:
+        logging.debug(
+            f"ARN for this resource is None, resource status is: {resource['status']}"
+        )
+    assert k8s.get_resource_arn(resource) is not None
+
+    yield (reference, spec, resource)
+    # Delete the k8s resource if not already deleted by tests
+    if k8s.get_resource_exists(reference):
+        _, deleted = k8s.delete_custom_resource(reference, 3, 10)
+        assert deleted
+
+
+@pytest.fixture(scope="function")
+def xgboost_unversioned_model_package():
+    resource_name = random_suffix_name("xgboost-unversioned-model-package", 38)
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["MODEL_PACKAGE_NAME"] = resource_name
+    reference, _, resource = create_sagemaker_resource(
+        resource_plural=RESOURCE_PLURAL,
+        resource_name=resource_name,
+        spec_file="xgboost_unversioned_model_package",
+        replacements=replacements,
+    )
+
+    assert resource is not None
+    if k8s.get_resource_arn(resource) is None:
```

Comment:
move 109 and 113 checks to the test, we only need this in fixtures where we are creating dependent resources (e.g. xgboost_model_package_group in this test) and are not the resource under test. Usually prefer failing test rather than a fixture

same for versioned